.gitignore

```diff
@@ -1,2 +1,13 @@
-/target
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```

Comment:
I think it might be better to keep a `Cargo.lock` within the repo since it ensures that when people build the code, it'll use the same versions of libraries we do and ensure that it compiles and runs correctly. Also if people wanted to update libraries on their own end while acknowledging that it could cause issues, they could run `cargo update`.

Reply:
Agreed